src/Jobs/Product/SaveProductJob.php

```diff
@@ -38,4 +41,23 @@ public function tags(): array
             $this->productData->identifier(),
         ];
     }
+
+    public function failed(Throwable $throwable): void
+    {
+        /** @var ?Product $model */
+        $model = Product::query()->firstWhere('identifier', '=', $this->productData->identifier());
+
+        $model?->failed();
+
+        activity()
+            ->when($model, function (ActivityLogger $logger, Product $product): ActivityLogger {
+                return $logger->on($product);
+            })
+            ->useLog('error')
+            ->withProperties([
+                'message' => $throwable->getMessage(),
+                'data' => $this->productData->toArray(),
+            ])
+            ->log('Failed to save the product data');
```

Comment:
Perhaps put the message of the exception here as well just like UpdateProductJob?